metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/utils/EntityMergeEngine.java

```diff
@@ -345,35 +346,34 @@ private static void convertAndAddEntity(RDF rdf, EnrichmentBase enrichmentBase,
       throw new IllegalArgumentException("Unknown entity type: " + enrichmentBase.getClass());
     }
 
-    // Append it to the proxy if needed.
-    if (entity != null && StringUtils.isNotEmpty(fieldName)) {
-      RdfProxyUtils.appendToProxy(rdf, entity, fieldName);
+    // Append it to the europeana proxy if needed.
+    if (entity != null && StringUtils.isNotBlank(fieldName)) {
+      RdfProxyUtils.appendToEuropeanaProxy(rdf, entity, fieldName);
     }
   }
 
   /**
-   * Merge entities in a record. Wrapper for {@link #mergeEntities(RDF, List, String)} without a
-   * field name.
+   * Merge entities in a record. Wrapper for {@link #mergeEntity(RDF, EnrichmentBase, String)}
+   * without a field name.
    *
    * @param rdf The RDF to enrich
-   * @param enrichmentBaseList The information to append
+   * @param enrichmentBaseWrapperList The information to append
    */
-  public void mergeEntities(RDF rdf, List<EnrichmentBase> enrichmentBaseList) {
-    mergeEntities(rdf, enrichmentBaseList, null);
+  public void mergeEntities(RDF rdf, List<EnrichmentBaseWrapper> enrichmentBaseWrapperList) {
+    for (EnrichmentBaseWrapper enrichmentBaseWrapper : enrichmentBaseWrapperList) {
+      mergeEntity(rdf, enrichmentBaseWrapper.getEnrichmentBase(),
+          enrichmentBaseWrapper.getOriginalField());
+    }
   }
 
   /**
-   * TODO JV This method is NEVER called with non-null (nonempty) field name? I would expect that
-   * particularly during enrichment (given the history of this method) there may be a need to
-   * provide this parameter.
-   *
    * Merge entities in a record
    *
    * @param rdf The RDF to enrich
-   * @param enrichmentBaseList The information to append
+   * @param enrichmentBase The information to append
    * @param fieldName The name of the field so that it can be connected to Europeana Proxy
    */
-  private void mergeEntities(final RDF rdf, List<EnrichmentBase> enrichmentBaseList,
+  private void mergeEntity(final RDF rdf, EnrichmentBase enrichmentBase,
```

Comment:
Why not pass the EnrichmentBaseWrapper? This would reduce the number of parameters by one.

Reply:
Makes sense, also changed another method down the chain.